### PR TITLE
Feb2024 Extended Sampling API Revision Tests

### PR DESCRIPTION
**Pull request by dougqh**
## Motivation

Updating Tests to include Feb 2024 revisions to the Extended Sampling API RFC
- glob case insensitivity
- glob extensions like set & range are forbidden
- glob special character handling
- matching of metrics in addition to meta

## Changes

Introduced test classes Test_Trace_Sampling_Globs_Feb2024_Revision & Test_Trace_Tag_Sampling_Feb2024_Revision

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
    * [ ] To R&P team: locally build and push the image to hub.docker.com 
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
    * [ ] Once merged, add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

File: tests/parametric/test_trace_sampling.py [hunk 1]
```diff
@@ -348,6 +394,175 @@ def test_trace_dropped_by_trace_sampling_rule_tags(self, test_agent, test_librar
...
+    @pytest.mark.parametrize("library_env", [tag_sampling_env("[a-c]")])
+    def test_no_range_support(self, test_agent, test_library):
+        """Test verifying that common glob range extension is NOT supported"""
+        with test_library:
+            with test_library.start_span(name="matching-span", service="test") as span:
+                span.set_meta("tag", "[a-c]")
+
+        matching_span = find_span_in_traces(
+            test_agent.wait_for_num_traces(1), Span(name="matching-span", service="test")
+        )
```
Comment (robertomonteromiguel) [inline tests/parametric/test_trace_sampling.py:474]: Could we extract these lines to a common method in the class? They are constantly repeated in the different methods, varying only: `span.set_meta("tag", "XYZ")`

File: tests/parametric/test_trace_sampling.py [hunk 2]
```diff
@@ -144,6 +145,51 @@ def test_trace_dropped_by_trace_sampling_rule(self, test_agent, test_library):
         assert span["metrics"].get(SAMPLING_RULE_PRIORITY_RATE) == 0.0
 
 
+@features.trace_sampling
+@scenarios.parametric
+@rfc("https://docs.google.com/document/d/1S9pufnJjrsxH6pRbpigdYFwA5JjSdZ6iLZ-9E7PoAic/")
+class Test_Trace_Sampling_Globs_Feb2024_Revision:
```
Comment (robertomonteromiguel) [inline tests/parametric/test_trace_sampling.py:151]: I advocate using another name instead of using dates. For example, the title of the rfc document

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]

File: tests/parametric/test_trace_sampling.py (hunk 1 above)
Comment (dougqh) [reply tests/parametric/test_trace_sampling.py:474]: I'd be fine with.  I debated whether or not that would obscure the test too much.

File: tests/parametric/test_trace_sampling.py (hunk 2 above)
Comment (dougqh) [reply tests/parametric/test_trace_sampling.py:151]: The issue here is that there were updates to the RFC as it was finalized.
Some tracers already implemented the first set, so I chose to split the tests into different sets of revisions.

**Review by robertomonteromiguel — APPROVED** [review]

**Pull request merged**